src/commonMain/kotlin/komem/litmus/LTDefinition.kt

```diff
@@ -30,11 +29,16 @@ class LTDefinitionScope<S>(
     }
 
     fun build(): LTDefinition<S> {
-        if (!::outcomeFinalizer.isInitialized) throw IllegalStateException("outcome not set")
+        val outcomeFinalizer: S.() -> LTOutcome = when {
+            ::outcomeFinalizer.isInitialized -> outcomeFinalizer
+            stateProducer() is AutoOutcome -> {
+                { (this as AutoOutcome).getOutcome() }
+            }
+
+            else -> error("outcome not specified")
+        }
```

Comment:
What is the idea here? 
The use can inherit its data class from e.g. `IIOutcome` to treat fields `r1` and `r2` as outcomes automatically?

Reply:
Yes, hence `AutoOutcome`